openslides_backend/action/actions/user/save_saml_account.py

```diff
@@ -0,0 +1,131 @@
+from typing import Any, Dict, Iterable, Sized, cast
+
+from openslides_backend.shared.util import ONE_ORGANIZATION_FQID
+
+from ....models.models import User
+from ....shared.exceptions import ActionException
+from ....shared.filters import FilterOperator
+from ....shared.interfaces.event import Event, EventType
+from ....shared.patterns import fqid_from_collection_and_id
+from ....shared.util import ONE_ORGANIZATION_ID
+from ...generics.create import CreateAction
+from ...mixins.send_email_mixin import EmailCheckMixin
+from ...util.action_type import ActionType
+from ...util.default_schema import DefaultSchema
+from ...util.register import register_action
+from ...util.typing import ActionData
+from .user_mixin import UsernameMixin
+
+
+@register_action("user.save_saml_account", action_type=ActionType.STACK_INTERNAL)
+class UserSaveSamlAccount(EmailCheckMixin, UsernameMixin, CreateAction):
+    """
+    Internal action to save (create or update) a saml account.
+    It should be called from the auth service.
+    """
+
+    check_email_field = "email"
+    model = User()
+    schema = DefaultSchema(User()).get_default_schema(
+        required_properties=["saml_id"],
+        optional_properties=[
+            "title",
+            "first_name",
+            "last_name",
+            "email",
+            "gender",
+            "pronoun",
+            "is_active",
+            "is_physical_person",
+        ],
+        title="create saml account schema",
+    )
+    skip_archived_meeting_check = True
+
+    def validate_instance(self, instance: Dict[str, Any]) -> None:
+        organization = self.datastore.get(
+            ONE_ORGANIZATION_FQID,
+            ["sso_enabled", "save_attr_config"],
+            lock_result=False,
+        )
+        if not organization.get("sso_enabled"):
+            raise ActionException(
+                "SingleSignOn is not enabled in OpenSlides configuration"
+            )
+        if not (attr_config := organization.get("save_attr_config")) or not isinstance(
+            attr_config, dict
+        ):
+            raise ActionException(
+                "SingleSignOn field attributes are not configured in OpenSlides"
+            )
+        new_instance = {
+            attr_config[key]: value
+            for key, value in instance.items()
+            if key in attr_config
+            and attr_config[key] in self.schema["properties"].keys()
+        }
+        instance.clear()
+        instance.update(new_instance)
```

Comment:
This was, because instance is only passed by ref, but not returned.
Extended the action framework to give a chance to modify the instance before validating the schema.